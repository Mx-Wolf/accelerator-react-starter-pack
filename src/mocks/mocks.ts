import {datatype, image, name, random} from 'faker';
import {Guitar} from '../types/guitar';

export const makeFakeComment = () => ({
  id: datatype.number(50).toString(),
  userName: name.title(),
  advantages: datatype.string(10),
  disadvantages: datatype.string(10),
  comment: datatype.string(10),
  rating: datatype.number(5),
  createAt: datatype.string(10),
  guitarId: datatype.number(5),
});

export const makeFakeCommentsList = (value: number) => new Array(value).fill(null).map(() => makeFakeComment());


export const makeFakeGuitar = () => ({
  comments: makeFakeCommentsList(4),
  id: datatype.number(50),
  name: name.title(),
  vendorCode: name.title(),
  type: random.word(),
  description: datatype.string(10),
  previewImg: image.image(),
  stringCount: datatype.number(12),
  rating: datatype.number(5),
  price: datatype.number(100),
});

export const makeFakeGuitarsList = (value: number): Guitar[] => new Array(value).fill(null).map(() => makeFakeGuitar());
