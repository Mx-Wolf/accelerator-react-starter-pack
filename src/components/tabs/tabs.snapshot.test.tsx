import renderer from 'react-test-renderer';
import Tabs from './tabs';
import {fakeGuitar} from '../../mocks/mocks';

describe('Component: Tabs', () => {
  it('should render correctly',  () => {
    const component = renderer.create(<Tabs productInfo={fakeGuitar} /> );
    expect(component).toMatchSnapshot();
  });
});
