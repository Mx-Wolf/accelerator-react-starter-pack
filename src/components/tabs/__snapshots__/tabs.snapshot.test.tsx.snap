// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tabs should render correctly 1`] = `
<div
  className="tabs"
>
  <a
    className="button  button--medium tabs__button"
    href="#characteristics"
    onClick={[Function]}
  >
    Характеристики
  </a>
  <a
    className="button button--black-border button--medium tabs__button"
    href="#description"
    onClick={[Function]}
  >
    Описание
  </a>
  <div
    className="tabs__content"
    id="characteristics"
  >
    <table
      className="tabs__table "
    >
      <tbody>
        <tr
          className="tabs__table-row"
        >
          <td
            className="tabs__title"
          >
            Артикул:
          </td>
          <td
            className="tabs__value"
          >
            code
          </td>
        </tr>
        <tr
          className="tabs__table-row"
        >
          <td
            className="tabs__title"
          >
            Тип:
          </td>
          <td
            className="tabs__value"
          />
        </tr>
        <tr
          className="tabs__table-row"
        >
          <td
            className="tabs__title"
          >
            Количество струн:
          </td>
          <td
            className="tabs__value"
          >
            2
             струнная
          </td>
        </tr>
      </tbody>
    </table>
    <p
      className="tabs__product-description visually-hidden"
    >
      description
    </p>
  </div>
</div>
`;
